reporter/src/main/kotlin/reporters/NoticeByPackageReporter.kt

```diff
@@ -0,0 +1,270 @@
+                    add { "\n" }
+                }
+
+                add { licenseText }
+
+                if (index < processedFindings.size - 1) {
+                    add { LICENSE_SEPARATOR }
+                }
+            }
+        }
+    }
+
+    private fun MutableList<() -> String>.addLicenseFileFindings(
+        licenseFileFindings: Map<String, LicenseFindingsMap>,
+        archiveDir: File
+    ) {
+        licenseFileFindings.forEach { (file, findings) ->
+            add { "This package contains the file $file with the following contents:\n\n" }
+            add { "${archiveDir.resolve(file).readText()}\n" }
+            val allCopyrights = findings.values.flatten().toSet()
+            val processedCopyrights = CopyrightStatementsProcessor().process(allCopyrights).toMutableSet()
+            if (processedCopyrights.isNotEmpty()) {
+                add { "The following copyright holder information relates to the license(s) above:\n\n" }
+            }
+            processedCopyrights.forEach { copyright ->
+                add { "$copyright\n" }
+            }
+        }
+    }
+
+    private fun getFindingsForLicenseFiles(
+        scanResult: ScanResult,
+        archiveDir: File,
+        licenseFindingsMap: LicenseFindingsMap
+    ): MutableMap<String, LicenseFindingsMap> {
+        val matcher = FileMatcher(LICENSE_FILENAMES)
+        val licenseFiles = mutableMapOf<String, LicenseFindingsMap>()
+
+        archiveDir.walkTopDown().forEach { file ->
+            val relativePath = archiveDir.toPath().relativize(file.toPath())
+            if (matcher.matches(relativePath.toString())) {
+                val licenses = findLicensesForFile(scanResult, relativePath)
+                val copyrightsInFile = findCopyrightsForFile(scanResult, relativePath)
+
+                val licensesWithCopyrights = licenses.associateWith { license ->
+                    // Remove copyrights which are already contained in the license file.
+                    val allCopyrights = (licenseFindingsMap[license].orEmpty() + copyrightsInFile).toSet()
+                    val processCopyrightsResult = CopyrightStatementsProcessor().process(allCopyrights)
+                    val processedCopyrightsNotInLicenseFile =
+                        processCopyrightsResult.processedStatements.filterNot { (_, sources) ->
+                            copyrightsInFile.any { it in sources }
+                        }
+                    val unprocessedCopyrightsNotInLicenseFile =
+                        processCopyrightsResult.unprocessedStatements - copyrightsInFile
+
+                    (processedCopyrightsNotInLicenseFile.keys + unprocessedCopyrightsNotInLicenseFile).toMutableSet()
+                }.toSortedMap()
+
+                licenseFiles[relativePath.toString()] = licensesWithCopyrights
+            }
+        }
+
+        return licenseFiles
+    }
+
+    private fun findCopyrightsForFile(scanResult: ScanResult, relativePath: Path): Set<String> {
```

Comment:
These two functions, `find*ForFile` introduce an additional code path which bypasses the matching of copyright findings to license findings and thus make things inconsistent. The implementation should be based on `OrtResult.collectLicenseFindings` instead. You could make the `private` function public for that purpose IMO.

Reply:
This is not a problem here, because these functions are only used to filter licenses and copyrights which appear in the license files for two reasons:

* To prevent licenses which appear in a license file from being listed again separately.
* To prevent listing copyrights below a license file which already appear in the license file.

Because we include license files completely, it is not important which license a copyright was associated to.

For all licenses that are listed separately the matched copyrights are used.

Reply:
oh, that makes sense. I completely missed that.